### ISSUE TITLE
Checkstyle is not run in GitHub CI

### ISSUE THREAD

**Issue by alanorth** [DSpace/DSpace#9038]
**Describe the bug**
Checkstyle is not run in GitHub CI. According to the comments in `.github/workflows/build.yml` it _should_ be run with the Unit Tests, but I noticed in [a recent PR](https://github.com/DSpace/DSpace/pull/9018) that it does not. This pull request introduced a [trailing space](https://github.com/DSpace/DSpace/pull/9018/files#diff-e8e68f2b1da2328f4846e0a4d34b39dedcc40453e9a3bc75eb1b5cd81cb37a1eR111), yet passed all sixteen checks in GitHub CI. When I ran `mvn -U checkstyle:check` locally, the trailing whitespace was detected and checkstyle threw an error.

**Expected behavior**
Checkstyle should be run in CI. We run lint in Angular CI so this should be no different.

**Comment by tdonohue** [comment]
@alanorth : Checkstyles are run by CI.  They run automatically (by default) whenever you run `mvn install`.   

The reason you see the [skipped notation](https://github.com/DSpace/DSpace/blob/main/.github/workflows/build.yml#L38) in that `build.yml` is because they are run already during the Unit Test phase.  See the comment here: https://github.com/DSpace/DSpace/blob/main/.github/workflows/build.yml#L30

So we skip them during Integration Tests in order to avoid running them twice for every CI build.

In other words, this runs checkstyle by default: https://github.com/DSpace/DSpace/blob/main/.github/workflows/build.yml#L71   That line triggers twice for every build.  Once for Unit Test (which runs checkstyle) and once for Integration Test (which disables checkstyle)

**Comment by tdonohue** [comment]
Proof that Checkstyle is running can be found in this PR: #9018 .  The second commit (the merge commit) threw 109 checkstyle errors in the "Run Unit Tests" phase: https://github.com/DSpace/DSpace/actions/runs/5782450021/job/15669286629
Those were then cleaned up and checkstyle passed.

Additional proof can be seen in the port PR #9037.... if you look at the logs of the "Run Unit Tests" phase, you'll see the "maven-checkstyle-plugin" is called for every module, e.g. here https://github.com/DSpace/DSpace/actions/runs/5782450021/job/15669286629

So, I suspect the trailing whitespace option may be either a bug in Checkstyle itself, or maybe two different versions of Checkstyle are in use (and one accepts trailing whitespace in comments and the other does not)

**Comment by tdonohue** [comment]
Closing, as this was just proven to work by @alanorth in this test PR to try to insert unallowable trailing whitespace.  https://github.com/DSpace/DSpace/pull/9040

So, Checkstyle is working.  But, it's still a mystery why it didn't seem to properly throw an error on the whitespace here https://github.com/DSpace/DSpace/pull/9018/files#diff-e8e68f2b1da2328f4846e0a4d34b39dedcc40453e9a3bc75eb1b5cd81cb37a1eR111

My best guess at this point is that maybe it's a bug in Checkstyle?
